.bazelrc

```diff
@@ -12,38 +12,51 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+###############################################################################
+# Common flags that apply to all configurations.
+# Use sparingly because they cannot be overriden.
+###############################################################################
+# Prevent invalid caching if input files are modified during a build.
+build --experimental_guard_against_concurrent_changes
+# Disable visibility checks (works around some private deps in TensorFlow that
+# are being unbundled soon anyway).
+build --nocheck_visibility
+build --compilation_mode=opt
+# Used in TensorFlow, so we have to enable it here as well.
+common --experimental_repo_remote_exec
+# Actually printing output on errors is... a useful default
+test --test_output=errors
+
+###############################################################################
+# Options for "generic_clang" builds: these options should generally apply to 
+# either clang or gcc and are curated based on need.
+###############################################################################
+
 # Disable warnings we don't care about.
-build --copt=-Wno-unused-local-typedef
-build --copt=-Wno-unused-private-field
-build --copt=-Wno-unused-const-variable
+build:generic_clang --copt=-Wno-unused-local-typedef
+build:generic_clang --copt=-Wno-unused-private-field
+build:generic_clang --copt=-Wno-unused-const-variable
 # Matches upstream. Cannot impact due to extern C inclusion method.
-build --copt=-Wno-extern-c-compat
+build:generic_clang --copt=-Wno-extern-c-compat
 # Technically UB but needed for intrusive ptrs
-build --copt=-Wno-invalid-offsetof
-build --copt=-Wno-unused-function
+build:generic_clang --copt=-Wno-invalid-offsetof
+build:generic_clang --copt=-Wno-unused-function
 
 # C++14 standard version is required.
-build --cxxopt=-std=c++14 --host_cxxopt=-std=c++14
+build:generic_clang --cxxopt=-std=c++14 --host_cxxopt=-std=c++14
 
 # Default to an optimized build with asserts.
 # This is a good compromise between runtime and debugability.
 # Override via: "-c dbg" or --compilation_mode=dbg
```

Comment:
The "Override" comment doesn't make sense here anymore. It's referring to `--compilation_mode=opt`

Reply:
Remove.